config/config.go

```diff
@@ -46,42 +46,55 @@ const Mainnet protocol.NetworkID = "mainnet"
 const GenesisJSONFile = "genesis.json"
 
 // Local holds the per-node-instance configuration settings for the protocol.
+// !!! WARNING !!!
+//
+// These versioned struct tags need to be maintained CAREFULLY and treated
+// like UNIVERSAL CONSTANTS - they should not be modified once committed.
+//
+// New fields may be added to the Local struct, along with a version tag
+// denoting a new version. When doing so, also update the
+// test/testdata/configs/config-v{n}.json and call "make generate" to regenerate the constants.
+//
+// !!! WARNING !!!
 type Local struct {
 	// Version tracks the current version of the defaults so we can migrate old -> new
 	// This is specifically important whenever we decide to change the default value
-	// for an existing parameter.
-	Version uint32
+	// for an existing parameter. This field tag must be updated any time we add a new version.
+	Version uint32 `version[0]:"0" version[1]:"1" version[2]:"2" version[3]:"3" version[4]:"4" version[5]:"5" version[6]:"6" version[7]:"7" version[8]:"8"`
```

Comment:
We need to be able to read this value off the current config file, and therefore it would helpful to keep this field here. As for the fact that this field has many entries - yes, we could make this field to have an exception where it would have a different struct tag which would denote the latest supported version only. I think that this could be broken into a separate PR though.

Reply:
Additionally `version[x]="0"` looks too verbose is case of multiple changes, maybe `v[x]="0"` ? Or even `0:"0"` ?

Reply:
With the exception of the version field, we rarely have more than two versions. I'd rather stick with code clarity over "compactness" here.